numpy/core/tests/test_multiarray.py

```diff
@@ -5507,6 +5507,39 @@ def test_var_values(self):
                 res = _var(mat, axis=axis)
                 assert_almost_equal(res, tgt)
 
+    @pytest.mark.parametrize(('complex_dtype', 'ndec'), (
+        ('complex64', 6),
+        ('complex128', 7),
+        ('clongdouble', 7),
+    ))
+    def test_var_complex_values(self, complex_dtype, ndec):
+        # Test fast-paths for every builtin complex type
+        for axis in [0, 1, None]:
+            mat = self.cmat.copy().astype(complex_dtype)
+            msqr = _mean(mat * mat.conj(), axis=axis)
+            mean = _mean(mat, axis=axis)
+            tgt = msqr - mean * mean.conjugate()
+            res = _var(mat, axis=axis)
+            assert_almost_equal(res, tgt, decimal=ndec)
+
+    def test_var_dimensions(self):
+        # _var paths for complex number introduce additions on views that
+        # increase dimensions. Ensure this generalizes to higher dims
```

Comment:
Note that this will make `var` fail on 32-dimensional arrays, since you'll try to insert another dimension which isn't allowed. I don't know if we care.

Reply:
We could add another condition of `arr.ndim < 32` to the fast-branch of the conditional if you think it's worth it. Is `32` hard-coded as the max dim in NumPy or is that a configurable value?

Reply:
Let's see what @seberg thinks. It's probably ok not to bother.

Reply:
Lets not bother, if you got 31 dimensions there are already a lot of other operations that break.